public_src/services/route-master-wizard.service.ts

```diff
@@ -0,0 +1,270 @@
+import { EventEmitter, Injectable } from '@angular/core';
+
+import { MapService } from './map.service';
+import { StorageService } from './storage.service';
+import { BsModalService } from 'ngx-bootstrap';
+import { ProcessService } from './process.service';
+
+import * as L from 'leaflet';
+import { IPtRelation } from '../core/ptRelation.interface';
+
+@Injectable()
+export class RouteMasterWizardService {
+  public map;
+  public routes              = [];
+  public ptLayerModal;
+  public osmtogeojson: any   = require('osmtogeojson');
+  public modalMapElementsMap = new Map();
+
+  public autoRouteMapNodeClick: EventEmitter<number>                    = new EventEmitter();
+  public newRoutesMapReceived: EventEmitter<Map<string, IPtRelation[]>> = new EventEmitter();
+
+  public savedMultipleNodeDataResponses = [];
+  public savedContinuousQueryResponses  = [];
+  public savedMasterQueryResponses      = [];
+
+  public elementsRenderedModalMap = new Set();
+  public nodesFullyDownloaded     = new Set();
+
+  // public routesMap: Map<string, any[]>                   = new Map();
+  public relsMap                                         = new Map();
+  public newRMsMap                                       = new Map();
+
+  constructor(private storageSrv: StorageService,
+              private mapSrv: MapService,
+              private modalService: BsModalService,
+              private processSrv: ProcessService) {
+    this.modalService.onShown.subscribe(() => {
+      this.onShownModal();
+    });
+    this.modalService.onHidden.subscribe(() => {
+      this.processAllDownloadedOnMainMap();
+      this.storageSrv.currentElement = null;
+      this.storageSrv.currentElementsChange.emit(
+        JSON.parse(JSON.stringify(null)),
+      );
+      this.storageSrv.stopsForRoute     = [];
+      this.storageSrv.platformsForRoute = [];
+      this.mapSrv.highlightType         = 'Stops';
+    });
+  }
+
+  /***
+   * Fired when modal has rendered
+   * @returns {void}
+   */
+  public onShownModal(): void {
+    if (this.map) {
+      this.map.invalidateSize();
+    }
+  }
+
+  /***
+   * Renders data on modal map which was already present on the main map
+   * @returns {void}
+   */
+  public renderAlreadyDownloadedData(): void {
+    let obj: any = {};
+    let elements = [];
```

Comment:
`const` because you are changing content and not the initialized object's reference...